pkg/server/server.go

```diff
@@ -382,8 +382,11 @@ func (s *Server) Run(ctx context.Context) error {
 			return err
 		}
 
-		// TODO(sttts): this is a hack, using the loopback config as a blueprint. Syncer should never use a loopback connection.
-		if err := s.installSyncerController(ctx, controllerConfig, CreateLoopbackUpstreamKubeConfig(server)); err != nil {
+		syncerConfig, err := s.options.AdminAuthentication.GetPushModeSyncerKubeconfig(genericConfig, newTokenOrEmpty, tokenHash, s.options.GenericControlPlane.GenericServerRunOptions.ExternalHost, servingOpts.BindPort)
```

Comment:
I think that ExternalAddress has logic to get always the right IP, 
```
	// ExternalAddress is the host name to use for external (public internet) facing URLs (e.g. Swagger)
	// Will default to a value based on secure serving info and available ipv4 IPs.
	ExternalAddress string
```

externalHost depends on be populated IIRC
```
staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go:  ExternalHost                string
staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go:  c.ExternalAddress = s.ExternalHost
staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go:  fs.StringVar(&s.ExternalHost, "external-hostname", s.ExternalHost,
```

Reply:
It iss already being used to generate the kubeconfig written to disk, so It has to be set.